File: api/src/main/java/org/delivery/api/domain/token/service/TokenService.java
package org.delivery.api.domain.token.service;

import lombok.RequiredArgsConstructor;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.delivery.api.domain.token.ifs.TokenHelperIfs;
import org.delivery.api.domain.token.model.TokenDto;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

@Service
@RequiredArgsConstructor
public class TokenService {

    private final TokenHelperIfs tokenHelperIfs;

    public TokenDto issueAccessToken(Long userId) {
        Map<String, Object> data = new HashMap<>();
        data.put("userId", userId);
        return tokenHelperIfs.issueAccessToken(data);
    }

    public TokenDto issueRefreshToken(Long userId) {
        Map<String, Object> data = new HashMap<>();
        data.put("userId", userId);
        return tokenHelperIfs.issueRefreshToken(data);
    }

    public Long validationToken(String token) {
        Map<String, Object> map = tokenHelperIfs.validationTokenWithThrow(token);

        Objects.requireNonNull(map.get("userId"), ()->{
            throw new ApiException(ErrorCode.NULL_POINT);
        });

        return Long.parseLong(map.get("userId").toString());
    }
}

File: account/src/main/kotlin/org/delivery/account/domain/token/service/TokenService.kt
package org.delivery.account.domain.token.service

import org.delivery.account.domain.token.ifs.TokenHelperIfs
import org.delivery.account.domain.token.model.TokenDto
import org.springframework.stereotype.Service

@Service
class TokenService(
    private val tokenHelperIfs: TokenHelperIfs
) {

    fun issueAccessToken(userId: Long?): TokenDto? {

        return userId?.let {
            val data = mapOf("userId" to it)
            tokenHelperIfs.issueAccessToken(data)
        }
    }

    fun issueRefreshToken(userId: Long?): TokenDto? {
        requireNotNull(userId)

        val data = mapOf("userId" to userId)
        return tokenHelperIfs.issueAccessToken(data)
    }

    fun validationToken(token: String?): Long? {
        return token?.let { token ->
            tokenHelperIfs.validationTokenWithThrow(token)
        }?.let { map ->
            map["userId"]
        }?.toString()?.toLong()

//        requireNotNull(token)
//
//        val map = tokenHelperIfs.validationTokenWithThrow(token)
//
//        val userId = map?.get("userId")
//
//        requireNotNull(userId)
//
//        return userId.toString().toLong()
    }

}